components/push/src/subscriber.rs

```diff
@@ -175,6 +188,108 @@ impl PushManager {
     }
 }
 
+// DB persisted rate limiter.
+// Implementation notes: This saves the timestamp of our latest call and the number of times we have
+// called `Self::check` within the `Self::periodic_interval` interval of time.
+pub struct PersistedRateLimiter {
+    op_name: String,
+    periodic_interval: u64, // In seconds.
+    max_requests_in_interval: u16,
+}
+
+impl PersistedRateLimiter {
+    pub fn new(op_name: &str, periodic_interval: u64, max_requests_in_interval: u16) -> Self {
+        Self {
+            op_name: op_name.to_owned(),
+            periodic_interval,
+            max_requests_in_interval,
+        }
+    }
```

Comment:
Seeing the rate-limiter refactor above led me to expect that you'd use that newly-refactored `RateLimiter` struct here, but you aren't. Is that because the algorithm that you want is different, or because it was hard to re-use when combined with this persistence abstraction?

Reply:
Yup, I realized too late that we needed a persisted counter versus an in-memory one :) Still worth refactoring though!